### PR TITLE
[MRG+1] Code removal for post 1.5/2.1

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by tacaswell** [comment]
This is code that is slated to be removed in 1.6/2.2 or beyond.

File: doc/api/api_changes/code_removal.rst [hunk 1]
```diff
@@ -4,3 +4,14 @@ Code Removal
 Legend
 ------
 Removed handling of `loc` as a positional argument to `Legend`
+
+
+matplotlib.delaunay
+-------------------
+Remove the delaunay triangulation code which is now handled by QHull
```
Comment (ianthomas23) [inline doc/api/api_changes/code_removal.rst]: QHull should be Qhull.

**Comment by ianthomas23** [comment]
Removal of delaunay looks good to me.

**Comment by tacaswell** [comment]
Addressed @ianthomas23 comment.  Rebased so it merges, but this is still on-hold until until the next major release.

**Comment by tacaswell** [comment]
@ianthomas23 Can you have another look at this if you still remember anything about this deprecation?

File: lib/matplotlib/tests/test_coding_standards.py [hunk 2]
```diff
@@ -203,7 +202,7 @@ def test_pep8_conformance_installed_files():
                           'tri/triinterpolate.py',
                           'tests/test_axes.py',
                           'tests/test_bbox_tight.py',
-                          'tests/test_delaunay.py',
+                          'tests/test_dviread.py',
```
Comment (Kojoley) [inline lib/matplotlib/tests/test_coding_standards.py]: Why adding this? `test_dviread.py` is not even touched.

Comment (tacaswell) [inline lib/matplotlib/tests/test_coding_standards.py]: failed merge conflict resolution.

File: doc/api/api_changes/code_removal.rst [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+matplotlib.delaunay
+-------------------
+Remove the delaunay triangulation code which is now handled by Qhull
+via ``matplotlib.tri``
+
+
+qt4_compat.py
+-------------
+Moved to ``qt_compat.py``.  Renamed because in now handles Qt5 as well.
```
Comment (NelleV) [inline doc/api/api_changes/code_removal.rst]: There is a small typo on this line: in -> it.

**Comment by ianthomas23** [comment]
@tacaswell I am happy for the delaunay code to be removed asap, it has been
deprecated for long enough.

**Pull request merged**